src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -78,7 +79,18 @@ trait BackendCall {
    * expression `read_lines(my_file_var)` would have to call lookupFunction()("my_file_var")
    * during expression evaluation
    */
-  def lookupFunction: String => WdlValue
+  def lookupFunction: String => WdlValue = {
+    def resolveParameter(name: String) = Try(locallyQualifiedInputs.getOrElse(name, throw new Throwable(s"variable $name not found")))
+    def resolveDeclaration(lookup: ScopedLookupFunction)(name: String) = key.scope.task.declarations.find(_.name == name) match {
+      case Some(d) => d.expression.map(_.evaluate(lookup, engineFunctions)).getOrElse(Failure(new WdlExpressionException(s"Could not evaluate declaration $d")))
+      case None => Failure(new Throwable(s"variable $name not found"))
+    }
+    def lookup(key: String): WdlValue = {
+      val attemptedResolutions = Stream(resolveParameter _, resolveDeclaration(lookupFunction) _) map {_(key)} find {_.isSuccess}
```

Comment:
Sorry, my explanation wasn't quite right.  While `find` should be lazy in all cases, `map` is strict on strict collections and would run all resolvers before invoking `find`:

```scala
scala> def f(x: Int) = { println("running f"); x } 
f: (x: Int)Int

scala> Stream(f _, f _) map { _(0) } find { _ == 0 } 
running f
res3: Option[Int] = Some(0)

scala> Seq(f _, f _) map { _(0) } find { _ == 0 } 
running f
running f
res4: Option[Int] = Some(0)
```